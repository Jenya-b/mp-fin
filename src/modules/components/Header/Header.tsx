import { useEffect, MouseEvent, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { Loader } from 'modules/components/Loader/Loader';
import { routerPath } from 'constants/routerPath';
import { useSignoutMutation } from 'services';
import { resetUser } from 'store/reducers/userSlice';
import { balanceSelector, userSelector } from 'store/selectors';
import { useAppDispatch, useAppSelector } from 'store/store';
import { HeaderDesktop } from './Desktop/Desktop';
import { HeaderMobile } from './Mobile/Mobile';
import { PullDownMenu } from '../PullDownMenu/PullDownMenu';
import { adminRoute, menuSidebar } from 'constants/menu';

export const Header = () => {
  const { balance, settings } = routerPath;
  const [isActiveMenu, setActiveMenu] = useState(false);
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const { user } = useAppSelector(userSelector);
  const currentBalance = useAppSelector(balanceSelector);

  const [signoutUser, { isSuccess, isLoading }] = useSignoutMutation();

  useEffect(() => {
    if (isSuccess) {
      dispatch(resetUser());
    }
  }, [dispatch, isSuccess]);

  const openPage = (event: MouseEvent<HTMLButtonElement>) => {
    const { name } = event.currentTarget;

    switch (name) {
      case balance:
        navigate(balance);
        break;
      case settings:
        navigate(settings);
        break;
      default:
        break;
    }
  };

  const onSignoutHandler = () => {
    signoutUser();
  };

  return (
    <>
      {isLoading && <Loader />}
      <HeaderDesktop
        user={user}
        balance={balance}
        settings={settings}
        currentBalance={currentBalance}
        openPage={openPage}
        onSignoutHandler={onSignoutHandler}
      />
      <HeaderMobile setActiveMenu={setActiveMenu} />
      <PullDownMenu
        dataMenu={menuSidebar}
        extraMenu={adminRoute}
        isActiveMenu={isActiveMenu}
        isAdmin={user?.isAdmin}
      />
    </>
  );
};
