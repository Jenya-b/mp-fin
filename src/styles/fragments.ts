import { css } from 'styled-components';

export const flexCenterAll = css`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const linkStyles = css`
  a {
    color: ${({ theme }) => theme.colors.formLink};

    :hover {
      color: ${({ theme }) => theme.colors.formLinkHover};
    }
  }
`;

export const ButtonMailStyles = css`
  border-radius: ${({ theme }) => theme.borders.primaryBtn.borderRadius}px;
  background: ${({ theme }) => theme.colors.btnbackground};
  transition: all ${({ theme }) => theme.transitions.button.time}s;

  :hover,
  :active {
    background: ${({ theme }) => theme.colors.btnHover};
  }
`;
