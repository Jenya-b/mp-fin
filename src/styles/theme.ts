import { ITheme } from 'interfaces/styled';
import { Colors } from 'constants/colors';

export const baseTheme: ITheme = {
  colors: {
    backgroundBase: Colors.MainBlue,
    backgroundPrimary: Colors.MainWhite,
    backgroundSecondary: Colors.MainGreenSecondary,
    backgroundTertiary: Colors.MainGreenPrimary,
    backgroundTelegramPrimary: Colors.AdditionalBlue,
    backgroundTelegramSecondary: Colors.AdditionalBlue,
    textBase: Colors.MainBlack,
    textPrimary: Colors.MainGreySecondary,
    textSecondary: Colors.MainGreyLight,
    textTertiary: Colors.MainGreyPrimary,
    textAttentionPrimary: Colors.MainBluePrimary,
    textAttentionSecondary: Colors.AdditionalPinkSecondary,
    borderPrimary: Colors.MainGreyLight,
    borderSecondary: Colors.MainGreenSecondary,
    borderTertiary: Colors.MainGreyTertiary,
  },
  sizes: {
    loginForm: {
      width: 390,
    },
    inputForm: {
      height: 55,
    },
    logoForm: {
      width: 590,
    },
    primaryBtn: {
      height: 55,
    },
    sidebar: {
      widthActive: 290,
      widthHidden: 90,
      logo: { height: 80 },
    },
    header: {
      height: 80,
    },
    footer: {
      height: 60,
    },
    notFound: {
      title: { fontSize: 56 },
    },
    errorFallback: {
      title: { fontSize: 56 },
    },
    settingInputFile: {
      width: 178,
      height: 178,
    },
    replenishmentBlock: {
      height: 239,
    },
    adminForm: {
      maxWidth: 900,
      width: 100,
    },
    telegramIcon: {
      width: 25,
      height: 25,
    },
  },
  indents: {
    loginForm: {
      paddingTop: 39,
      paddingRight: 45,
      paddingBottom: 39,
      paddingLeft: 45,
    },
    inputList: {
      marginTop: 25,
      rowGap: 15,
    },
    main: {
      paddingLeft: 290,
      paddingLeftHide: 90,
    },
    adminMenu: {
      paddingTop: 40,
      paddingRight: 25,
      paddingBottom: 40,
      paddingLeft: 30,
    },
    adminList: {
      rowGap: 10,
    },
    adminContainer: {
      marginTop: 30,
    },
    adminForm: {
      columnGap: 10,
    },
    adminSubtitle: {
      marginBottom: 10,
    },
    adminSearchBlock: {
      marginTop: 30,
    },
    queryTableRow: {
      paddingTop: 7,
      paddingBottom: 7,
    },
  },
  borders: {
    loginForm: { borderRadius: 10 },
    inputForm: { borderRadius: 10 },
    primaryBtn: { borderRadius: 10 },
    replenishment: { borderRadius: 10 },
  },
  transitions: {
    input: { time: 0.3 },
    button: { time: 0.3 },
  },
  // z-index
  order: {
    sidebar: 50,
    settingInputFile: 10,
    settingLogo: 1,
  },
  cursor: 'pointer',
  media: {
    extraLarge: '(max-width: 1279px)',
    large: '(max-width: 1023px)',
    medium: '(max-width: 767px)',
    small: '(max-width: 479px)',
  },
};
