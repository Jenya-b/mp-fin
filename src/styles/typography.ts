import { css } from 'styled-components';

const fontsFamilyOpenSans = css`
  font-family: 'Open Sans', sans-serif;
  font-style: normal;
`;

const fontsFamilyInter = css`
  font-family: 'Inter', sans-serif;
  font-style: normal;
`;

export const fontStylesH1 = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 28px;
  line-height: 38px;
`;
export const fontStylesH2 = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 22px;
  line-height: 30px;
`;
export const fontStylesRegularBold = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
`;
export const fontStylesRegular = css`
  ${fontsFamilyOpenSans}
  font-weight: 400;
  font-size: 18px;
  line-height: 25px;
`;
export const fontStylesCaptionBig = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
`;
export const fontStylesCaption = css`
  ${fontsFamilyInter}
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
`;
export const fontStylesSmall = css`
  ${fontsFamilyInter}
  font-weight: 400;
  font-size: 12px;
  line-height: 22px;
`;
