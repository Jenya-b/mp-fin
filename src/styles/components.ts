import styled from 'styled-components';
import { ButtonMailStyles } from 'styles/fragments';
import {
  fontStylesCaption,
  fontStylesH1,
  fontStylesH2,
  fontStylesRegularBold,
} from 'styles/typography';

export const Main = styled.main`
  padding: 80px 60px;
`;

export const MainTitle = styled.h1`
  ${fontStylesH1}
  color: ${({ theme }) => theme.colors.mainTitle};
`;

export const MainSubtitle = styled.h2`
  ${fontStylesH2}
  color: ${({ theme }) => theme.colors.mainTitle};
`;

export const Button = styled.button`
  color: ${({ theme }) => theme.colors.btn};
  ${ButtonMailStyles}
`;

export const PrimaryButton = styled(Button)`
  width: 100%;
  height: ${({ theme }) => theme.sizes.primaryBtn.height}px;
  ${fontStylesRegularBold}
`;

export const Input = styled.input`
  width: 100%;
  height: ${({ theme }) => theme.sizes.inputForm.height}px;
  padding-left: 20px;
  ${fontStylesCaption}
  color: ${({ theme }) => theme.colors.inputFormText};
  border-radius: ${({ theme }) => theme.borders.inputForm.borderRadius}px;
  border: 1px solid ${({ theme }) => theme.colors.inputFormBorder};
  transition: all ${({ theme }) => theme.transitions.input.time}s;

  ::placeholder {
    color: ${({ theme }) => theme.colors.inputFormPlaceholder};
  }

  :hover {
    border: 1px solid ${({ theme }) => theme.colors.inputFormHoverBord};
  }

  :focus {
    border: 1px solid ${({ theme }) => theme.colors.inputFormFocusBord};
  }
`;

export const PrimaryInput = styled(Input)`
  background: ${({ theme }) => theme.colors.inputFormBackground};
`;

export const SearchInput = styled(Input)`
  background: ${({ theme }) => theme.colors.inputFormBackground};
`;

export const Checkbox = styled.input``;
