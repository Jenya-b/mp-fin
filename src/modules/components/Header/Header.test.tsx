import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { baseTheme } from 'styles/theme';
import { Header } from './Header';
import {
  StyledHeader,
  BalanceButton,
  BalanceIcon,
  BalanceInfo,
  BalanceSum,
  ButtonLogin,
  ButtonSettings,
  ButtonWrapper,
  Controls,
  LoginImage,
  LoginInfo,
  LoginName,
  LoginTitle,
} from './Header.styled';

describe('Snapshot Header styled-components', () => {
  test('StyledHeader', () => {
    const tree = renderer.create(<StyledHeader theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('BalanceButton', () => {
    const tree = renderer.create(<BalanceButton theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('BalanceIcon', () => {
    const tree = renderer.create(<BalanceIcon theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('BalanceInfo', () => {
    const tree = renderer.create(<BalanceInfo theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('BalanceSum', () => {
    const tree = renderer.create(<BalanceSum theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('ButtonLogin', () => {
    const tree = renderer.create(<ButtonLogin theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('ButtonSettings', () => {
    const tree = renderer.create(<ButtonSettings theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('ButtonWrapper', () => {
    const tree = renderer.create(<ButtonWrapper theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Controls', () => {
    const tree = renderer.create(<Controls theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('LoginImage', () => {
    const tree = renderer
      .create(<LoginImage imagesUrl={'/source/images/iconAccount.png'} theme={baseTheme} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('LoginInfo', () => {
    const tree = renderer.create(<LoginInfo theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('LoginName', () => {
    const tree = renderer.create(<LoginName theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('LoginTitle', () => {
    const tree = renderer.create(<LoginTitle theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
