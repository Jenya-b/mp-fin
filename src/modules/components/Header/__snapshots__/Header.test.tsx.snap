// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Header styled-components BalanceButton 1`] = `
<button
  className="sc-hLBbgP sc-bqWxrE dYfrRk fnJgIk"
/>
`;

exports[`Snapshot Header styled-components BalanceIcon 1`] = `
<div
  className="sc-csuSiG ACVId"
/>
`;

exports[`Snapshot Header styled-components BalanceInfo 1`] = `
<div
  className="sc-ipEyDJ dVJfLL"
/>
`;

exports[`Snapshot Header styled-components BalanceSum 1`] = `
<div
  className="sc-eDWCr gsVzSY"
/>
`;

exports[`Snapshot Header styled-components ButtonLogin 1`] = `
<button
  className="sc-hBxehG sc-bjfHbI eJsBfT hcFlRF"
/>
`;

exports[`Snapshot Header styled-components ButtonSettings 1`] = `
<button
  className="sc-hBxehG sc-fEXmlR eJsBfT hjSRwe"
/>
`;

exports[`Snapshot Header styled-components ButtonWrapper 1`] = `
<div
  className="sc-fnGiBr kuUJiF"
/>
`;

exports[`Snapshot Header styled-components Controls 1`] = `
<div
  className="sc-ksBlkl eFVbIr"
/>
`;

exports[`Snapshot Header styled-components LoginImage 1`] = `
<div
  className="sc-jrcTuL bvSNSN"
/>
`;

exports[`Snapshot Header styled-components LoginInfo 1`] = `
<div
  className="sc-pyfCe cuSZop"
/>
`;

exports[`Snapshot Header styled-components LoginName 1`] = `
<div
  className="sc-kDvujY foeDiA"
/>
`;

exports[`Snapshot Header styled-components LoginTitle 1`] = `
<div
  className="sc-idXgbr eVbslW"
/>
`;

exports[`Snapshot Header styled-components StyledHeader 1`] = `
<header
  className="sc-ftTHYK hmjFDk"
/>
`;
